src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -260,7 +260,9 @@ public final void apply(Project project) {
         // we can't just have the task run the write locks code as we need to write the locks in afterEvaluate.
         project.getTasks()
                 .register(WRITE_VERSIONS_LOCKS_TASK, WriteVersionsLocksMarkerTask.class, writeVersionsLocks -> {
-                    writeVersionsLocks.getOutputs().upToDateWhen(_ignored -> false);
+                    writeVersionsLocks.getDirectlyTriggeredByHuman().set(VersionsLockPlugin.shouldWriteLocks(project));
+                    // upToDateWhen(_ignored -> false) is not configuration cacheable use doNotTrackState instead
+                    writeVersionsLocks.doNotTrackState("This task should always run.");
```

Comment:
Fwiw `doNotTrackState` is apparently more efficient than `upToDateWhen(false)`, because it turns off all state tracking so it doesn't even bother to compare to the current and previous states. It might be worth adding this to the list of errorprones as an autofix, would be a straightforward one.

Reply:
You can also apparently annotate the task with `@UntrackedTask("reason")` which might be more discoverable.